src/utils.rs

```diff
@@ -253,8 +253,10 @@ pub fn parse_batch(msg:&Batch)-> Ruddermessage{
     for i in &msg.batch {
         match i {
             BatchMessage::Identify(a_) =>{
-                let mut final_context: Value = a_.context.clone().unwrap_or(json!({}));
-                merge(&mut final_context, modified_context.clone() );
+                let mut event_context: Value = a_.context.clone().unwrap_or(json!({}));
+                let mut final_context: Value = batch_context.clone();
+                merge(&mut event_context, default_context.clone());
+                merge(&mut final_context, event_context.clone());
```

Comment:
This should be the correct implementation.

- Event-level context takes precedence over batch-level context
- Default context takes precedence over any custom context

```suggestion
                let mut event_context: Value = a_.context.clone().unwrap_or(json!({}));
                let mut final_context: Value = batch_context.clone();
                merge(&mut final_context, event_context);
                merge(&mut final_context, default_context.clone());
```

Reply:
Addressed

Reply:
No we don't have unit test case setup